app/src/main/java/com/example/bigowlapp/MonitoringGroupListActivity.java

```diff
@@ -0,0 +1,149 @@
+    private ListView users_listview;
+    private TextView groupName, supervisorName;
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private List<User> mUsers, mUsersShow;
+    private List<String> mSupervisedGroup;
+
+    EditText search_users;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_monitoring_group_list);
+        initialize();
+    }
+
+    protected void initialize() {
+        mUsers = new ArrayList<>();
+        mSupervisedGroup = new ArrayList<>();
+
+        try {
+            FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
+            if (currentUser != null) {
+
+                db.collection("groups")
+                        .whereEqualTo("monitoringUserId", currentUser.getUid())
+                        .get()
+                        .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                            @Override
+                            public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                                if (task.isSuccessful()) {
+                                    for (QueryDocumentSnapshot qds : task.getResult()) {
+
+                                        Group g = qds.toObject(Group.class);
+                                        groupName = findViewById(R.id.textView_groupName);
+                                        groupName.setText(qds.getString("name"));
+
+
+                                        for (String supervisedUser : g.getSupervisedUserId()) {
+                                            mSupervisedGroup.add(supervisedUser);
+                                        }
+                                        //find Supervisor full name
+                                        db.collection("users").document(qds.getString("monitoringUserId")).get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
+                                            @Override
+                                            public void onComplete(@NonNull Task<DocumentSnapshot> task) {
+                                                if (task.isSuccessful()) {
+                                                    User u = task.getResult().toObject(User.class);
+                                                    supervisorName = findViewById(R.id.textView_supervisorName);
+                                                    supervisorName.setText(u.toString());
+                                                }
+                                            }
+                                        });
+                                    }
+
+                                    //List of all user to compare with
+                                    db.collection("users")
+                                            .whereIn(FieldPath.documentId(), mSupervisedGroup)
```

Comment:
![image](https://user-images.githubusercontent.com/23643192/96390283-eaa3f880-1181-11eb-9784-9dfd45f8d8aa.png)

This relates to the issue when a new user clicks on the monitoring group button.